### PR TITLE
Add mutual TLS support to prometheus_client output plugin

### PR DESCRIPTION
**Pull request by robertjsullivan**
Signed-off-by: Robert Sullivan <rsullivan@pivotal.io>

### Required for all PRs:

- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [X] Has appropriate unit tests.

### CODE REVIEW

**Comment by danielnelson** [comment]
This would be a nice feature, we actually have most of this implemented for use in plugins.  The best example to follow is in `http_listener_v2`:

https://github.com/influxdata/telegraf/blob/cf18c4a2bffab89ca62c8333dca867397e39359d/plugins/inputs/http_listener_v2/http_listener_v2.go#L37

For tests, look in https://github.com/influxdata/telegraf/tree/master/testutil/pki

File: plugins/outputs/prometheus_client/prometheus_client_tls_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
+package prometheus_client_test
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/influxdata/telegraf/plugins/outputs/prometheus_client"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/influxdata/toml"
+	. "github.com/onsi/gomega"
```
Comment (danielnelson) [inline plugins/outputs/prometheus_client/prometheus_client_tls_test.go]: We will need to remove this dependency, I don't want to add it only for testing this one aspect.

File: plugins/outputs/prometheus_client/prometheus_client.go [hunk 2]
```diff
@@ -105,6 +106,12 @@ var sampleConfig = `
   ## If set, enable TLS with the given certificate.
   # tls_cert = "/etc/ssl/telegraf.crt"
   # tls_key = "/etc/ssl/telegraf.key"
+  
+  ## If set, enable TLS client authentication with the given CA.
+  # tls_ca = "/etc/ssl/telegraf_ca.crt"
```
Comment (danielnelson) [inline plugins/outputs/prometheus_client/prometheus_client.go]: Update this option:
```
  ## Set one or more allowed client CA certificate file names to 
  ## enable mutually authenticated TLS connections
  # tls_allowed_cacerts = ["/etc/telegraf/clientca.pem"]
```

File: plugins/outputs/prometheus_client/prometheus_client.go [hunk 3]
```diff
@@ -105,6 +106,12 @@ var sampleConfig = `
...
+
+  ## Boolean value indicating whether or not to skip SSL verification
+  # insecure_skip_verify = false
```
Comment (danielnelson) [inline plugins/outputs/prometheus_client/prometheus_client.go]: Remove this option, it's not available for the server configuration.  Don't forget to update the README as well.

File: plugins/outputs/prometheus_client/prometheus_client_tls_test.go (hunk 1 above)
Comment (kitemongerer) [reply plugins/outputs/prometheus_client/prometheus_client_tls_test.go]: Replaced with require

File: plugins/outputs/prometheus_client/prometheus_client.go (hunk 2 above)
Comment (kitemongerer) [reply plugins/outputs/prometheus_client/prometheus_client.go]: Done

File: plugins/outputs/prometheus_client/prometheus_client.go (hunk 3 above)
Comment (kitemongerer) [reply plugins/outputs/prometheus_client/prometheus_client.go]: Done

**Pull request merged**